src/Expand.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace Sarhan;
+
+/**
+ * @author  Frank Koornstra
+ * @license LGPL
+ */
+class Expand
+{
+    /**
+     * Expands a one-dimensional array that has flattened keys to a (possibly) multi-dimensional array
+     *
+     * @param array  $array     The array with flattened keys to expand
+     * @param string $separator The separator used in the keys when it was flattened
+     * @return array A possibly multi dimensional array
+     * @throws \RuntimeException
+     */
+    public static function inflate($array, $separator = '.')
+    {
+        if (! is_array($array)) {
+            throw new \RuntimeException('Cannot expand non-arrays');
+        }
+
+        $splitPattern = '/' . preg_quote($separator, '/') . '/';
+
+        $inflated = [];
+        foreach ($array as $key => $value) {
+            $parent = &$inflated;
+
+            $partList = preg_split($splitPattern, $key, -1, PREG_SPLIT_NO_EMPTY);
```

Comment:
i'm sorry (😬 ) but this is the kind of case where `$parts` would be much simpler & clearer that `$partList`, IMO. `$partList` makes it sounds like it's a specific data structure or something (especially when used in a context where we have data structure terms like "leaf").